### PR TITLE
Refactor/app structure

### PR DESCRIPTION
**Pull request by yechielb2000**
## Huge change! Please go through this one.

### CODE REVIEW

**Comment by yechielb2000** [comment]
We can merge and if there is a problem I will deal with it myself.

File: frontend/lib/doc_types/orot_user.dart
Comment (itamarshubin) [inline frontend/lib/doc_types/orot_user.dart:1]: Yes, I moved it to provider

File: frontend/lib/models/district_model.dart [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// use JsonSerializable
```
Comment (itamarshubin) [inline frontend/lib/models/district_model.dart:1]: Add TODO:

File: frontend/lib/doc_types/orot_user.dart
Comment (yechielb2000) [reply frontend/lib/doc_types/orot_user.dart:1]: I see that `FirestoreService` uses this but I don't see that you use `FirestoreService`.
Do you still need [this](https://github.com/itamarshubin/Orot/blob/Dev/frontend/lib/services/firestore_service.dart)? if not we can delete `OrotUser` and `FirestoreService`.

File: frontend/lib/models/district_model.dart (hunk 1 above)
Comment (yechielb2000) [reply frontend/lib/models/district_model.dart:1]: I imlemented it.

**Pull request merged**